types/src/transaction/transaction_target.rs

```diff
@@ -0,0 +1,236 @@
+use alloc::vec::Vec;
+use core::fmt::{self, Debug, Display, Formatter};
+
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+#[cfg(any(feature = "testing", test))]
+use rand::{Rng, RngCore};
+#[cfg(feature = "json-schema")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[cfg(doc)]
+use super::Transaction;
+use super::{TransactionInvocationTarget, TransactionRuntime, TransactionSessionKind};
+use crate::bytesrepr::{self, Bytes, FromBytes, ToBytes, U8_SERIALIZED_LENGTH};
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
+
+const NATIVE_TAG: u8 = 0;
+const STORED_TAG: u8 = 1;
+const SESSION_TAG: u8 = 2;
+
+/// The execution target of a [`Transaction`].
+#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(
+    feature = "json-schema",
+    derive(JsonSchema),
+    schemars(description = "Execution target of a Transaction.")
+)]
+#[serde(deny_unknown_fields)]
+pub enum TransactionTarget {
```

Comment:
Good point!  I think `Entity` isn't as appropriate since it could represent either an addressable entity or a package.

@EdHastingsCasperAssociation - I'll merge as is, but if you object I'll rename the variant in a separate PR.